modules/navier_stokes/src/physics/WCNSLinearFVTurbulencePhysics.C

```diff
@@ -0,0 +1,345 @@
+  params.set<LinearVariableName>("variable") = _tked_name;
+  if (shouldCreateTimeDerivative(_tked_name, _blocks, /*error if already defined*/ false))
+    getProblem().addLinearFVKernel(kernel_type, prefix() + "tked_time", params);
+}
+
+void
+WCNSLinearFVTurbulencePhysics::addKEpsilonAdvection()
+{
+  const std::string kernel_type = "LinearFVTurbulentAdvection";
+  InputParameters params = getFactory().getValidParams(kernel_type);
+
+  assignBlocks(params, _blocks);
+
+  params.set<UserObjectName>("rhie_chow_user_object") = _flow_equations_physics->rhieChowUOName();
+  params.set<MooseEnum>("advected_interp_method") =
+      getParam<MooseEnum>("tke_advection_interpolation");
+  params.set<LinearVariableName>("variable") = _tke_name;
+  getProblem().addLinearFVKernel(kernel_type, prefix() + "tke_advection", params);
+  params.set<LinearVariableName>("variable") = _tked_name;
+  params.set<std::vector<BoundaryName>>("walls") = _turbulence_walls;
+  params.set<MooseEnum>("advected_interp_method") =
+      getParam<MooseEnum>("tked_advection_interpolation");
+  getProblem().addLinearFVKernel(kernel_type, prefix() + "tked_advection", params);
+}
+
+void
+WCNSLinearFVTurbulencePhysics::addKEpsilonDiffusion()
+{
+  {
+    const std::string kernel_type = "LinearFVTurbulentDiffusion";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    assignBlocks(params, _blocks);
+    params.set<bool>("use_nonorthogonal_correction") =
+        getParam<bool>("use_nonorthogonal_correction");
+
+    params.set<LinearVariableName>("variable") = _tke_name;
+    params.set<MooseFunctorName>("diffusion_coeff") =
+        _flow_equations_physics->dynamicViscosityName();
+    getProblem().addLinearFVKernel(kernel_type, prefix() + "tke_diffusion_mu", params);
+
+    params.set<std::vector<BoundaryName>>("walls") = _turbulence_walls;
+    params.set<LinearVariableName>("variable") = _tked_name;
+    getProblem().addLinearFVKernel(kernel_type, prefix() + "tked_diffusion_mu", params);
+  }
+
+  {
+    const std::string kernel_type = "LinearFVTurbulentDiffusion";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    assignBlocks(params, _blocks);
+    params.set<bool>("use_nonorthogonal_correction") =
+        getParam<bool>("use_nonorthogonal_correction");
+
+    params.set<LinearVariableName>("variable") = _tke_name;
+    params.set<MooseFunctorName>("diffusion_coeff") = _turbulent_viscosity_name;
+    params.set<MooseFunctorName>("scaling_coeff") = getParam<MooseFunctorName>("sigma_k");
+    getProblem().addLinearFVKernel(kernel_type, prefix() + "tke_diffusion_mu_turb", params);
+
+    params.set<std::vector<BoundaryName>>("walls") = _turbulence_walls;
+    params.set<LinearVariableName>("variable") = _tked_name;
+    params.set<MooseFunctorName>("scaling_coeff") = getParam<MooseFunctorName>("sigma_eps");
+    getProblem().addLinearFVKernel(kernel_type, prefix() + "tked_diffusion_mu_turb", params);
+  }
```

Comment:
We should have only one block `mu + mu_t/sigma`. Linear FV doesn't work very well with the definition of multiple kernels for the same physics. If you define a flux BC it will be added twice because of this.

Reply:
so if we use an auxkernel for mu_eff = mu_t + mu I think we would not use the boundary condition

Reply:
ok so we go the functor material route

Reply:
all 3. a little bit